builder/builder.go

```diff
@@ -208,7 +208,7 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 				shouldPush := opt.Push && saveImage.Push && !sts.Target.IsRemote() && saveImage.DockerTag != ""
 				shouldExport := !opt.NoOutput && opt.OnlyArtifact == nil && !(opt.OnlyFinalTargetImages && sts != mts.Final) && saveImage.DockerTag != ""
 				useCacheHint := saveImage.CacheHint && b.opt.CacheExport != ""
-				if !shouldPush && !shouldExport && !useCacheHint {
+				if (!shouldPush && !shouldExport && !useCacheHint) || b.builtMain {
```

Comment:
This line is the meat of the PR

Reply:
This works until I add my PR, #754

Does this also fix that?